packages/swing-store/src/sqlStreamStore.js

```diff
@@ -90,14 +90,8 @@ export function sqlStreamStore(dbDir, io) {
       Fail`${q(startPosition.itemCount)} <= ${q(endPosition.itemCount)}}`;
 
     function* reader() {
-      ensureTransaction();
-      const query = db.prepare(`
-        SELECT item
-        FROM streamItem
-        WHERE streamName = ? AND position >= ? AND position < ?
-        ORDER BY position
-      `);
-      for (const { item } of query.iterate(
+      ensureTxn();
+      for (const { item } of sqlReadStreamQuery.iterate(
```

Comment:
> I'm sure this was a problem with previous version too, but am I correct in thinking that this iterator wouldn't be happy with other DB txns happening in the middle of iteration?

Yes, the code here is essentially unchanged aside from the `ensureTxn` call and some renaming.

> Our current use case for reading the stream is vat transcript replay, and that doesn't do a lot of other DB activity (syscalls are handled from the transcript, not executed for real), but Mathieu's #6664 XS/GC bug workaround involves allowing certain `vatstoreGet` calls to be executed during replay, and those would cause DB reads. We aren't landing that on trunk, but it suggests that we shouldn't depend too deeply on excluding other DB activity during sequential reads of a list like this.

This should not be a problem, as reads do not interfere with reads.  The issue is trying to do an `UPDATE` while in the midst of a `SELECT`.  In this case that should not ever happen.